modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -135,6 +135,8 @@
     private DiagnosticLog dlog;
     private TypeChecker typeChecker;
     private Stack<WorkerActionSystem> workerActionSystemStack = new Stack<>();
+    private Stack<Boolean> retryStmtCheckStack = new Stack<>();
+    private Stack<BLangStatement> loopWithintransactionCheckStack = new Stack<>();
```

Comment:
This can be done by having a Boolean stack, which says, we are inside a loop construct (true) or inside a transaction block (false).

Reply:
Fixed with f6fe91e29e529078b0c173726ea698ee63e9fe62. Thanks